sound/soc/sof/Kconfig

```diff
@@ -166,6 +166,15 @@ config SND_SOC_SOF_DEBUG_RETAIN_DSP_CONTEXT
 	  Say Y if you want to retain DSP context for FW exceptions.
 	  If unsure, select "N".
 
+config SND_SOC_SOF_DEBUG_PROBES
+	bool "SOF enable probes"
+	select SND_SOC_COMPRESS
+	help
+	  This option enables the data probing feature that can be used to
+	  gather data directly from specific points of the audio pipeline.
+	  Say Y if you want to enable probes.
+	  If unsure, select "N".
+
```

Comment:
Question: do we want to also have the ability to enable probes with a kernel parameter, which adds more functionality? e.g. we can force the traces to be on, or we can enable them dynamically.

Reply:
I decided to run with general DEBUG_PROBES kconfig and then a separate ones for system specific implementations. For Intel that's SND_SOC_SOF_HDA_PROBES.